agent/agent.go

```diff
@@ -89,7 +95,12 @@ func NewAgent(config *Config) (*Agent, error) {
 
 	// We don't check for errors here and let the agent get created regardless
 	// of system info status.
-	_ = agent.refreshSystemInfo()
+	systemInfoCtx, cancel := context.WithTimeout(ctx, time.Duration(DefaultSystemInfoRefreshInterval)*time.Second)
+	defer cancel()
+	_ = agent.refreshSystemInfo(systemInfoCtx)
+	if err := systemInfoCtx.Err(); err != nil {
```

Comment:
`systemInfoCtx.Err()` returns the same error as `agent.refreshSystemInfo(systemInfoCtx)`?

Reply:
They can be different. Here, I'm checking systemInfoCtx.Err() to figure out if the context has been canceled.